esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -1427,39 +1452,22 @@ void EsdlServiceImpl::prepareFinalRequest(IEspContext &context,
     // error checking has already been completed.
 
-    if (!serviceCrt)
-    {
-        if (m_serviceLevelCrtFail)
-            throw MakeStringException(-1, "%s::%s disabled due to service-level Custom Transform errors. Review transform template in configuration.", srvdef.queryName(), mthName);
-
-        serviceCrt = m_serviceLevelRequestTransform;
-    }
+    if (m_serviceLevelCrtFail)
+        throw MakeStringException(-1, "%s::%s disabled due to service-level Custom Transform errors. Review transform template in configuration.", srvdef.queryName(), mthName);
 
-    if (!methodCrt)
-    {
-        StringAttr *crtErrorMessage = m_methodCRTransformErrors.getValue(mthName);
-        if (crtErrorMessage && !crtErrorMessage->isEmpty())
-            throw MakeStringException(-1, "%s::%s disabled due to method-level Custom Transform errors: %s. Review transform template in configuration.", srvdef.queryName(), mthName, crtErrorMessage->str());
+    m_serviceLevelRequestTransform;
 
-        methodCrt = m_customRequestTransformMap.getValue(mthName);
-    }
+    StringAttr *crtErrorMessage = m_methodCRTransformErrors.getValue(mthName);
+    if (crtErrorMessage && !crtErrorMessage->isEmpty())
+        throw MakeStringException(-1, "%s::%s disabled due to method-level Custom Transform errors: %s. Review transform template in configuration.", srvdef.queryName(), mthName, crtErrorMessage->str());
 
-    if (serviceCrt || methodCrt)
+    if (scriptContext)
     {
-        context.addTraceSummaryTimeStamp(LogNormal, "srt-custreqtrans");
+        IEsdlCustomTransform *methodCrt = m_customRequestTransformMap.getValue(mthName);
 
-        //as we add more entry points the script context will move to wider scope
-        Owned<IEsdlScriptContext> scriptContext = createEsdlScriptContext(&context);
+        context.addTraceSummaryTimeStamp(LogNormal, "srt-custreqtrans");
         scriptContext->setContent(ESDLScriptCtxSection_ESDLRequest, reqProcessed.str());
-        scriptContext->setAttribute(ESDLScriptCtxSection_ESDLInfo, "service", srvdef.queryName());
-        scriptContext->setAttribute(ESDLScriptCtxSection_ESDLInfo, "method", mthdef.queryMethodName());
-        scriptContext->setAttribute(ESDLScriptCtxSection_ESDLInfo, "request_type", mthdef.queryRequestType());
-        scriptContext->setAttribute(ESDLScriptCtxSection_ESDLInfo, "request", mthdef.queryRequestType());  //this could diverge from request_type in the future
-
-        scriptContext->setContent(ESDLScriptCtxSection_TargetConfig, tgtcfg);
-        scriptContext->setContent(ESDLScriptCtxSection_BindingConfig, m_oEspBindingCfg.get());
-
-        processServiceAndMethodTransforms(scriptContext, {serviceCrt, methodCrt}, ESDLScriptCtxSection_ESDLRequest, ESDLScriptCtxSection_FinalRequest);
+        processServiceAndMethodTransforms(scriptContext, {m_serviceLevelRequestTransform, methodCrt}, ESDLScriptCtxSection_ESDLRequest, ESDLScriptCtxSection_FinalRequest);
```

Comment:
This should not be called if both transforms are NULL (it wasn't - see old line 1447), or this method should be updated to recognize when all initializer list values are NULL. There's a lot of overhead involved preparing to process no transforms.

Reply:
If there are no transforms there will currently be no scriptContext.  But I may readd the check just so the overhead doesn't reappear if in the future we create scriptContext for some other reason and there are no CRTs.

Reply:
My expectation has been that service, when it takes ownership of the log content blob, should create the same content structure with or without defined transforms (for any or all entry points) so the next entry point (including the log update entry point in the agent) don't have to figure out what happened before it. I would err on the side of caution and avoid the overhead.